cmake/mission_build.cmake

```diff
@@ -326,29 +355,67 @@ endfunction(prepare)
 
   # Note - A warning is issued if you pass CMAKE_TOOLCHAIN_FILE to an already-configured build area
   # so an extra check is added to see if this is an initial run or a re-run by checking for a CMakeCache file.
-  if (NOT CURRSYS STREQUAL "native" AND NOT EXISTS "${ARCH_BINARY_DIR}/CMakeCache.txt")
+  if (NOT ARCH_TOOLCHAIN_NAME STREQUAL "native" AND NOT EXISTS "${ARCH_BINARY_DIR}/CMakeCache.txt")
     # Find the toolchain file - allow a file in the mission defs dir to supercede one in the compile dir
-    if (EXISTS ${MISSION_DEFS}/toolchain-${CURRSYS}.cmake)
-      set(TOOLCHAIN_FILE ${MISSION_DEFS}/toolchain-${CURRSYS}.cmake)
-    elseif(EXISTS ${CFE_SOURCE_DIR}/cmake/toolchain-${CURRSYS}.cmake)
-      set(TOOLCHAIN_FILE ${CFE_SOURCE_DIR}/cmake/toolchain-${CURRSYS}.cmake)
+    if (EXISTS "${MISSION_DEFS}/toolchain-${ARCH_TOOLCHAIN_NAME}.cmake")
+      set(TOOLCHAIN_FILE "${MISSION_DEFS}/toolchain-${ARCH_TOOLCHAIN_NAME}.cmake")
+    elseif(EXISTS "${CFE_SOURCE_DIR}/cmake/toolchain-${ARCH_TOOLCHAIN_NAME}.cmake")
+      set(TOOLCHAIN_FILE "${CFE_SOURCE_DIR}/cmake/toolchain-${ARCH_TOOLCHAIN_NAME}.cmake")
     else()
-      message(FATAL_ERROR "Unable to find toolchain file for ${CURRSYS}")
+      message(FATAL_ERROR "Unable to find toolchain file for ${ARCH_TOOLCHAIN_NAME}")
     endif()
     set(SELECTED_TOOLCHAIN_FILE "-DCMAKE_TOOLCHAIN_FILE=${TOOLCHAIN_FILE}")
   else()
     # Do not supply any toolchain file option to the subprocess
     set(SELECTED_TOOLCHAIN_FILE)
   endif ()
   
+  # Generate wrapper file for the requisite cfe_platform_cfg.h file
+  generate_config_includefile(
+    OUTPUT_DIRECTORY    "${ARCH_BINARY_DIR}/inc" 
+    FILE_NAME           "cfe_msgids.h"
+    MATCH_SUFFIX        "msgids.h"
+    PREFIXES            ${BUILD_CONFIG} ${ARCH_TOOLCHAIN_NAME}
+  )
+  generate_config_includefile(
+    OUTPUT_DIRECTORY    "${ARCH_BINARY_DIR}/inc"
+    FILE_NAME           "cfe_platform_cfg.h" 
+    MATCH_SUFFIX        "platform_cfg.h" 
+    PREFIXES            ${BUILD_CONFIG} ${ARCH_TOOLCHAIN_NAME}
+  )
+  
+  # Create wrappers for any files provided in an "fsw/platform_inc" subdirectory of each app
+  # These may be optionally overridden by providing a file of the same name in the defs directory
+  foreach(APP_NAME ${MISSION_APPS})
+    set(APP_PLATFORM_CONFIG_DIR "${${APP_NAME}_MISSION_DIR}/fsw/platform_inc")
+    if (IS_DIRECTORY "${APP_PLATFORM_CONFIG_DIR}")
+        file(GLOB APP_PLATFORM_CONFIG_FILES RELATIVE "${APP_PLATFORM_CONFIG_DIR}" "${APP_PLATFORM_CONFIG_DIR}/*.h")
+        foreach(CONFIG_FILE ${APP_PLATFORM_CONFIG_FILES})            
+            generate_config_includefile(
+                OUTPUT_DIRECTORY    "${ARCH_BINARY_DIR}/inc" 
+                FILE_NAME           "${CONFIG_FILE}" 
+                FALLBACK_FILE       "${APP_PLATFORM_CONFIG_DIR}/${CONFIG_FILE}" 
+                MATCH_SUFFIX        "${CONFIG_FILE}"
+                PREFIXES            ${BUILD_CONFIG} ${ARCH_TOOLCHAIN_NAME}
```

Comment:
Instead of prefixing, what about a subdirectory in the defs directory? I can imagine a lot of overrides once every app's _msgids.h file is overridden w/ a database output for every CPU.